TrafficCapture/captureOffloader/src/test/java/org/opensearch/migrations/trafficcapture/StreamChannelConnectionCaptureSerializerTest.java

```diff
@@ -315,8 +331,56 @@ public void testEndOfSegmentsIndicationNotAddedWhenNotChunking() throws IOExcept
         Assertions.assertEquals(0, foundEndOfSegments);
     }
 
+    @Test
+    public void testAssertionErrorDuringInitializationWhenInitializeWithTooLargeId() {
+        final String realNodeId = "b671d2f2-577b-414e-9eb4-8bc3e89ee182";
+        final String realKafkaConnectionId = "9a25a4fffe620014-00034cfa-00000001-d208faac76346d02-864e38e2";
+
+        var outputBuffersCreated = new ConcurrentLinkedQueue<ByteBuffer>();
+        assertThrows(AssertionError.class, () ->
+                new StreamChannelConnectionCaptureSerializer<>("a" + realNodeId, realKafkaConnectionId,
+                        new StreamManager(getEstimatedTrafficStreamByteSize(0, 0), outputBuffersCreated))
+        );
+    }
+
+    @Test
+    public void testInitializationWithRealIds() {
+        final String realNodeId = "b671d2f2-577b-414e-9eb4-8bc3e89ee182";
+        final String realKafkaConnectionId = "9a25a4fffe620014-00034cfa-00000001-d208faac76346d02-864e38e2";
+
+        var outputBuffersCreated = new ConcurrentLinkedQueue<ByteBuffer>();
+        new StreamChannelConnectionCaptureSerializer<>(realNodeId, realKafkaConnectionId,
+                new StreamManager(getEstimatedTrafficStreamByteSize(0, 0), outputBuffersCreated));
+    }
+
+    @Test
+    public void testOutputStreamReportsIncorrectSpaceLeft_thenObservationSizeSanityCheckLogAppears() throws IOException {
+        Logger mockLogger = Mockito.mock(Logger.class);
+
+        StreamManager mockStreamManager = Mockito.mock(StreamManager.class);
+        CodedOutputStreamHolder mockHolder = Mockito.mock(CodedOutputStreamHolder.class);
+        CodedOutputStream mockOutputStream = Mockito.mock(CodedOutputStream.class);
+
+        Mockito.when(mockStreamManager.createStream()).thenReturn(mockHolder);
+        Mockito.when(mockHolder.getOutputStream()).thenReturn(mockOutputStream);
+        Mockito.when(mockOutputStream.spaceLeft()).thenReturn(5);
+        var serializer = new StreamChannelConnectionCaptureSerializer<>(TEST_NODE_ID_STRING, TEST_TRAFFIC_STREAM_ID_STRING,
+                mockStreamManager, mockLogger);
+        var bb = Unpooled.buffer(getEstimatedTrafficStreamByteSize(1, 0));
+        serializer.addWriteEvent(Instant.now(), bb);
+        serializer.flushCommitAndResetStream(true);
+        bb.release();
+
+        Mockito.verify(mockLogger, Mockito.times(1)).warn(
```

Comment:
For What's it's it worth - there's some code in a MetricsLogger test that rigs a Log4J appender - then you don't need to monkey with mockito.

Reply:
Thanks